contracts/contracts/Tally.sol

```diff
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.10;
+
+import {AccQueue} from "./trees/AccQueue.sol";
+import {IMACI} from "./IMACI.sol";
+import {Hasher} from "./crypto/Hasher.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {Poll} from "./Poll.sol";
+import {MessageProcessor} from "./MessageProcessor.sol";
+import {SnarkCommon} from "./crypto/SnarkCommon.sol";
+import {Verifier} from "./crypto/Verifier.sol";
+import {VkRegistry} from "./VkRegistry.sol";
+
+
+contract Tally is
+    Ownable,
+    SnarkCommon,
+    Hasher
+{
+    // Error codes
+    string constant ERROR_VOTING_PERIOD_NOT_PASSED = "TallyE01";
```

Comment:
I would say it might be worth to migrating to custom errors throughout the code base, and remove all require statements for if checks. What you think? should help save some gas

Reply:
i.e.

```js
error VOTING_PERIOD_NOT_OVER();
error PROCESSING_NOT_COMPLETE();
error INVALID_TALLY_VOTES_PROOF();
error ALL_BALLOTS_TALLIED();
```

Reply:
Sounds good. But let's put it into the next PR to separate refactor and optimization.